nestkernel/connection.h

```diff
@@ -134,7 +136,7 @@ class Connection
    * To prevent erronous calls of this function on primary connections, the base class implementation
    * below just contains `assert(false)`.
    */
-  SecondaryEvent* get_secondary_event();
+  [[noreturn]] SecondaryEvent* get_secondary_event();
```

Comment:
The method is overridden in derived classes implementing `SecondaryEvent`s, and there it returns a pointer. I assume there is no problem with having the `[[noreturn]]` attribute in the base class but not in the derived classes?

Reply:
I don't think this attribute is inherited by derived classes (I did try to look at the standard documents that I have access to). Looking at https://en.cppreference.com/w/cpp/language/attributes/noreturn I would expect that this only affects the "one" function where it was declared.